### PR TITLE
feat(dms): new check `dms_endpoint_ssl_enabled`

### LINKED ISSUE #4878: Add new check `dms_endpoint_use_ssl`
### New feature motivation

SSL/TLS connections provide encryption for data in transit between DMS replication instances and your databases. This ensures that the data is protected from eavesdropping and tampering during migration. Enabling SSL also includes certificate validation, which verifies that connections are made to the intended database instance by checking its server certificate.

Reference:
https://docs.aws.amazon.com/securityhub/latest/userguide/dms-controls.html#dms-9

### Solution Proposed

Create a new check to ensure that DMS endpoints is using SSL. Things to add:
- Add `endpoints` dictionary with all endpoints extracted from [describe_endpoints](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dms/client/describe_endpoints.html) and test it. The key of dict must be the endpoint ARN.
- Add check logic to verify that endpoint's `SslMode` is not 'none'. [Ref](https://docs.prowler.com/projects/prowler-open-source/en/latest/developer-guide/checks/)
- Add unit tests to ensure the check is working as expected. Cases recommended: no endpoints, endpoint without SSL active, endpoint with SSL required, endpoint with SSL in other modes. [Ref](https://docs.prowler.com/projects/prowler-open-source/en/latest/developer-guide/unit-testing/)

### Describe alternatives you've considered

No alternative.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by LefterisXefteris**
### Context

Ensure SSL mode is enabled when creating a DMS endpoint

Fix #4878

### Description

This check ensures that SSL mode is enabled for all AWS Database Migration Service (DMS) endpoints. Enabling SSL provides encryption in transit for data transferred through these endpoints

### License
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -4962,4 +4962,4 @@ test = ["big-O", "importlib-resources", "jaraco.functools", "jaraco.itertools",
 [metadata]
 lock-version = "2.0"
 python-versions = ">=3.9,<3.13"
-content-hash = "c6eb860b261fd62646806a6abb6a41a0296a452a7fb6c1bda8da715aeeb41124"
+content-hash = "d010e8fd2962654e3bcf44e8a463de4d4a5cbfc4009f46649faea6ac6f40a6f0"
```
Comment (puchy22) [inline poetry.lock]: I believe that this change should not appear when the `mkdocs` dependency is removed. Please execute `poetry lock --no-update` when `pyproject` is correct to synchronize this file as well.

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "Provider": "aws",
+  "CheckID": "dms_endpoint_ssl_enabled",
+  "CheckTitle": "Ensure SSL mode is enabled when creating a DMS endpoint",
+  "CheckType": [
+      "Data Protection",
+      "Encryption in Transit"
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]: This type should be extracted from: https://docs.aws.amazon.com/securityhub/latest/userguide/asff-required-attributes.html#Types

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "Provider": "aws",
+  "CheckID": "dms_endpoint_ssl_enabled",
+  "CheckTitle": "Ensure SSL mode is enabled when creating a DMS endpoint",
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
  "CheckTitle": "Ensure SSL mode is enabled in DMS endpoint",
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      "Encryption in Transit"
+  ],
+  "ServiceName": "dms",
+  "SubServiceName": "endpoint",
+  "ResourceIdTemplate": "arn:partition:dms:region:account-id:endpoint:resource-id",
+  "Severity": "high",
+  "ResourceType": "AwsDmsEndpoint",
+  "Description": "This check ensures that SSL mode is enabled for all AWS Database Migration Service (DMS) endpoints. Enabling SSL provides encryption in transit for data transferred through these endpoints.",
+  "Risk": "Without SSL enabled, data transferred through DMS endpoints is not encrypted, potentially exposing sensitive information to unauthorized access or interception during transit.",
+  "RelatedUrl": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.html#CHAP_Security.SSL",
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
  "RelatedUrl": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.SSL.html",
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  "Remediation": {
+      "Code": {
+          "CLI": "aws dms modify-endpoint --endpoint-arn <endpoint_arn> --ssl-mode require",
+          "NativeIaC": "",
+          "Other": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.html#CHAP_Security.SSL.Configuring",
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
          "Other": "https://docs.aws.amazon.com/securityhub/latest/userguide/dms-controls.html#dms-9",
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+          "Terraform": "resource \"aws_dms_endpoint\" \"example\" {\n  # ... other configurations ...\n  ssl_mode = \"require\"\n}"
+      },
+      "Recommendation": {
+          "Text": "Enable SSL mode for all DMS endpoints. Use 'require' as the minimum SSL mode, and consider using 'verify-ca' or 'verify-full' for higher security.",
+          "Url": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.html#CHAP_Security.SSL"
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
          "Url": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.SSL.html"
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  "ResourceType": "AwsDmsEndpoint",
+  "Description": "This check ensures that SSL mode is enabled for all AWS Database Migration Service (DMS) endpoints. Enabling SSL provides encryption in transit for data transferred through these endpoints.",
+  "Risk": "Without SSL enabled, data transferred through DMS endpoints is not encrypted, potentially exposing sensitive information to unauthorized access or interception during transit.",
+  "RelatedUrl": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.html#CHAP_Security.SSL",
+  "Remediation": {
+      "Code": {
+          "CLI": "aws dms modify-endpoint --endpoint-arn <endpoint_arn> --ssl-mode require",
+          "NativeIaC": "",
+          "Other": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.html#CHAP_Security.SSL.Configuring",
+          "Terraform": "resource \"aws_dms_endpoint\" \"example\" {\n  # ... other configurations ...\n  ssl_mode = \"require\"\n}"
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
          "Terraform": ""
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  "CheckTitle": "Ensure SSL mode is enabled when creating a DMS endpoint",
+  "CheckType": [
+      "Data Protection",
+      "Encryption in Transit"
+  ],
+  "ServiceName": "dms",
+  "SubServiceName": "endpoint",
+  "ResourceIdTemplate": "arn:partition:dms:region:account-id:endpoint:resource-id",
+  "Severity": "high",
+  "ResourceType": "AwsDmsEndpoint",
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json:13]:
```suggestion
  "ResourceType": "AwsDmsEndpointDetails",
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      },
+      "Recommendation": {
+          "Text": "Enable SSL mode for all DMS endpoints. Use 'require' as the minimum SSL mode, and consider using 'verify-ca' or 'verify-full' for higher security.",
+          "Url": "https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.html#CHAP_Security.SSL"
+      }
+  },
+  "Categories": [
+      "encryption",
+      "security-best-practices"
+  ],
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
  "Categories": [
      "encryption"
  ],
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  "DependsOn": [],
+  "RelatedTo": [
+      "dms_replication_instance_not_public"
+  ],
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
  "RelatedTo": [],
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  "Notes": "Ensure that the SSL certificates are properly configured and managed for the source and target endpoints."
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
  "Notes": ""
```

File: tests/providers/aws/services/dms/dms_instance_no_public_access/dms_no_public_access_test.py [hunk 12]
```diff
@@ -24,7 +24,7 @@ def test_dms_no_instances(self):
 
         from prowler.providers.aws.services.ec2.ec2_service import EC2
 
-        aws_provider = set_mocked_aws_provider([AWS_REGION_US_EAST_1])
+        aws_provider = set_mocked_aws_provider([AWS_REGION_US_EAST_1])
```
Comment (puchy22) [inline tests/providers/aws/services/dms/dms_instance_no_public_access/dms_no_public_access_test.py]:
```suggestion
        aws_provider = set_mocked_aws_provider([AWS_REGION_US_EAST_1])
```

File: tests/providers/aws/services/dms/dms_instance_multi_az/dms_instance_multi_az_test.py [hunk 13]
```diff
@@ -24,7 +24,7 @@ def test_dms_no_instances(self):
             )
 
             check = dms_instance_multi_az_enabled()
-            result = check.execute()
+            result = check.execute()
```
Comment (puchy22) [inline tests/providers/aws/services/dms/dms_instance_multi_az/dms_instance_multi_az_test.py]:
```suggestion
            result = check.execute()
```

File: pyproject.toml [hunk 14]
```diff
@@ -71,6 +71,7 @@ shodan = "1.31.0"
 slack-sdk = "3.31.0"
 tabulate = "0.9.0"
 tzlocal = "5.2"
+mkdocs = "1.5.3"
```
Comment (puchy22) [inline pyproject.toml]:
```suggestion
```

This dependency is already added when you use `poetry install --with docs`

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.py [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+            if endpoint.ssl_mode == 'none':
+                report.status = "FAIL"
+                report.status_extended = f"DMS Endpoint {endpoint_id} is not using SSL."
+            else:
+                report.status = "PASS"
+                report.status_extended = f"DMS Endpoint {endpoint_id} is using SSL with mode: {endpoint.ssl_mode}."
+            
+            findings.append(report)
+            
+        return findings
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.py]:
```suggestion
        return findings

```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 16]
```diff
@@ -1,5 +1,5 @@
 from pydantic import BaseModel
-
+from typing import List, Optional, Dict
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
```

Why adding this if they are not being used?

File: prowler/providers/aws/services/dms/dms_service.py [hunk 17]
```diff
@@ -11,7 +11,9 @@ def __init__(self, provider):
         # Call AWSService's __init__
         super().__init__(__class__.__name__, provider)
         self.instances = []
+        self.endpoints = {}
         self.__threading_call__(self.__describe_replication_instances__)
+        self.__threading_call__(self.__describe_endpoints__)
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
        self.__threading_call__(self._describe_replication_instances)
        self.__threading_call__(self._describe_endpoints)
```
Please, add testing for the new method in `dms_service_test.py`

File: prowler/providers/aws/services/dms/dms_service.py [hunk 18]
```diff
@@ -11,7 +11,9 @@ def __init__(self, provider):
...
 
     def __describe_replication_instances__(self, regional_client):
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
    def _describe_replication_instances(self, regional_client):
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 19]
```diff
@@ -49,6 +51,31 @@ def __describe_replication_instances__(self, regional_client):
                 f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
+    def __describe_endpoints__(self, regional_client):
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
    def _describe_endpoints(self, regional_client):
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 20]
```diff
@@ -49,6 +51,31 @@ def __describe_replication_instances__(self, regional_client):
...
+        logger.info("DMS - Describing DMS Endpoints...")
+        try:
+            describe_endpoints_paginator = regional_client.get_paginator(
+                "describe_endpoi4nts"
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
                "describe_endpoints"
```
Have you verified that the check works in AWS?

File: prowler/providers/aws/services/dms/dms_service.py [hunk 21]
```diff
@@ -49,6 +51,31 @@ def __describe_replication_instances__(self, regional_client):
...
+                        )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+
+
+class Endpoint(BaseModel):
+    ssl_mode:str
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
class Endpoint(BaseModel):
    id: str
    ssl_mode: str
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 22]
```diff
@@ -49,6 +51,31 @@ def __describe_replication_instances__(self, regional_client):
...
+                "describe_endpoi4nts"
+            )
+            for page in describe_endpoints_paginator.paginate():
+                for endpoint in page["Endpoints"]:
+                    arn = endpoint["EndpointArn"]
+                    if not self.audit_resources or (
+                        is_resource_filtered(arn, self.audit_resources)
+                    ):
+                        self.endpoints[endpoint["EndpointIdentifier"]] = Endpoint(
+                            ssl_mode=endpoint.get("SslMode")
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
                        self.endpoints[arn] = Endpoint(
                            id=endpoint["EndpointIdentifier"]
                            ssl_mode=endpoint.get("SslMode", False)
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.py [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+from prowler.providers.aws.services.dms.dms_client import dms_client
+
+class dms_endpoint_ssl_enabled(Check):
+    def execute(self):
+        findings = []
+        for endpoint_id, endpoint in dms_client.endpoints.items():
+            report = Check_Report_AWS(self.metadata())
+            report.resource_id = endpoint_id
+            report.resource_arn = f"arn:{dms_client.audited_partition}:dms:{dms_client.audited_region}:{dms_client.audited_account}:endpoint:{endpoint_id}"
+            report.region = dms_client.audited_region
```
Comment (puchy22) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.py]:
```suggestion
        for endpoint_arn, endpoint in dms_client.endpoints.items():
            report = Check_Report_AWS(self.metadata())
            report.resource_id = endpoint.id
            report.resource_arn = endpoint_arn
            report.region = dms_client.audited_region
```

File: tests/providers/aws/services/dms/dms_endpoint_ssl_enabled_test/dms_endpoint_ssl_enabled_test.py [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+            "prowler.providers.aws.services.dms.dms_service.DMS",
+            new=dms_client,
+        ):
+            from prowler.providers.aws.services.dms.dms_endpoint_ssl_enabled.dms_endpoint_ssl_enabled import dms_endpoint_ssl_enabled
+
+            check = dms_endpoint_ssl_enabled()
+            result = check.execute()
+            assert len(result) == 0
+
+    def test_dms_endpoint_ssl_none(self):
```
Comment (puchy22) [inline tests/providers/aws/services/dms/dms_endpoint_ssl_enabled_test/dms_endpoint_ssl_enabled_test.py:25]: Please adapt all MagickMocks to new Endpoint Model using as key the ARN

**Review by puchy22 — DISMISSED** [review]
Hi @LefterisXefteris, this is an awesome job 💯. Please review my suggested changes.

File: prowler/providers/aws/services/dms/dms_service.py (hunk 20 above)
Comment (LefterisXefteris) [reply prowler/providers/aws/services/dms/dms_service.py]: Hi, i am testing this on AWS now

Comment (LefterisXefteris) [reply prowler/providers/aws/services/dms/dms_service.py]: sorry i realised this. this must be a typo be before i make my first commit. i remember, i spent some time time figuring out how to get_paginate() method works and how ti fetches the endpoint(as im dealing with dicts)

File: prowler/providers/aws/services/dms/dms_service.py [hunk 25]
```diff
@@ -1,5 +1,4 @@
 from pydantic import BaseModel
-
 from prowler.lib.logger import logger
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion

from prowler.lib.logger import logger
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 26]
```diff
@@ -49,6 +52,66 @@ def _describe_replication_instances(self, regional_client):
...
+            describe_data_provider_paginator = regional_client.get_paginator(
+                "describe_data_providers"
+            )
+
+            for page in describe_data_provider_paginator.paginate():
+                for provider in page["DataProviders"]:
+                    name = provider['DataProviderName']
+                    if not self.audit_resources or (
+                        is_resource_filtered(name, self.audit_resources)
+                    ):
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
                    arn = provider['DataProviderArn']
                    if not self.audit_resources or (
                        is_resource_filtered(arn, self.audit_resources)
                    ):
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 27]
```diff
@@ -49,6 +52,66 @@ def _describe_replication_instances(self, regional_client):
...
+                        settings = provider.get('Settings', {})
+                        for setting_key, setting_value in settings.items():
+                            ssl_mode = setting_value.get('SslMode')
+                            if ssl_mode:
+                                settings[setting_value] = ssl_mode
+                            
+                        self.data_providers.append(DataProvider(
+                            name=DataProvider['DataProviderName'],
+                            settings=DataProvider["Settings"]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
                        self.data_providers.append(DataProvider(
                            arn=arn,
                            name=DataProvider['DataProviderName'],
                            settings=DataProvider["Settings"]
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 28]
```diff
@@ -49,6 +52,66 @@ def _describe_replication_instances(self, regional_client):
...
+            describe_endpoints_paginator = regional_client.get_paginator(
+                "describe_endpoints"
+            )
+            for page in describe_endpoints_paginator.paginate():
+                for endpoint in page["Endpoints"]:
+                    arn = endpoint["EndpointArn"]
+                    if not self.audit_resources or (
+                        is_resource_filtered(arn, self.audit_resources)
+                    ):
+                        self.endpoints[endpoint["EndpointIdentifier"]] = Endpoint(
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
                        self.endpoints[arn] = Endpoint(
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 29]
```diff
@@ -49,6 +52,66 @@ def _describe_replication_instances(self, regional_client):
...
+                            settings=DataProvider["Settings"]
+                        ))
+                            
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+class DataProvider(BaseModel):
+    name: str
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
class DataProvider(BaseModel):
    arn: str
    name: str
```

File: prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json [hunk 30]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "dms_endpoint_ssl_enabled",
+  "CheckTitle": "Ensure SSL mode is enabled in DMS endpoint",
+  "CheckType": ["Data Exposure"],
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_endpoint_ssl_enabled/dms_endpoint_ssl_enabled.metadata.json]:
```suggestion
  "CheckType": ["Effects", "Data Exposure"],
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 31]
```diff
@@ -49,6 +52,68 @@ def _describe_replication_instances(self, regional_client):
...
+                            id=endpoint["EndpointIdentifier"],
+                            ssl_mode=endpoint.get("SslMode", False)
+                        )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+
+    def _describe_data_providers(self, regional_client):
```
Comment (sergargar) [inline prowler/providers/aws/services/dms/dms_service.py:77]: Hi @LefterisXefteris , why did you create `_describe_data_providers` function?

Comment (sergargar) [reply prowler/providers/aws/services/dms/dms_service.py:77]: Moreover, could you give us access to the branch so we can help you better with the changes? Thanks!

File: prowler/providers/aws/services/dms/dms_service.py [hunk 32]
```diff
@@ -49,6 +52,68 @@ def _describe_replication_instances(self, regional_client):
...
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+class DataProvider(BaseModel):
+    arn: str
+    name: str
+    settings: dict
+
```
Comment (sergargar) [inline prowler/providers/aws/services/dms/dms_service.py:111]:
```suggestion
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 33]
```diff
@@ -49,6 +52,68 @@ def _describe_replication_instances(self, regional_client):
...
+                            
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+class DataProvider(BaseModel):
+    arn: str
+    name: str
+    settings: dict
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 34]
```diff
@@ -11,7 +11,10 @@ def __init__(self, provider):
         # Call AWSService's __init__
         super().__init__(__class__.__name__, provider)
         self.instances = []
+        self.endpoints = {}
+        self.data_providers = []
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
```

**Review by MrCloudSec — APPROVED** [review]
Thanks for your contribution @LefterisXefteris !

**Pull request merged**